php_amqp.h

```diff
@@ -298,8 +184,9 @@ extern zend_class_entry *amqp_exception_class_entry,
 		zend_throw_exception(amqp_connection_exception_class_entry, verify_connection_error_tmp, 0 TSRMLS_CC); \
 		return; \
 
+// Adjust for offset
 #define AMQP_VERIFY_CONNECTION(connection, error) \
-	if (!connection) { \
+	if (!connection || connection > -1000) { \
```

Comment:
That's a hack that needs to be fixed.

As cou can see in amqp_connection_object_fetch_object the amqp_connection_object* now is at an offset from the zend_object*. Because the struct changed (because zend_object don't have a fixed size anymore and need to be end the end of objects). That means that when zend_object* is null, the connection you get via amqp_connection_object_fetch_object is not null, but at 0-offset.